nameko/standalone/rpc.py

```diff
@@ -64,14 +65,27 @@ class PollingQueueConsumer(object):
     separate thread it provides a polling method to block until a message with
     the same correlation ID of the RPC-proxy call arrives.
     """
+    consumer = None
+
     def __init__(self, timeout=None):
         self.timeout = timeout
+        self.replies = {}
+
+    def _setup_consumer(self):
+        if self.consumer is not None:
+            try:
+                self.consumer.cancel()
+            except socket.error:  # pragma: no cover
+                pass  # already disconnected
```

Comment:
Might be worth a note here for posterity, explaining that some platforms set ``self.consumer`` to ``None`` on disconnection and some don't -- so it's not possible to reasonably cover this.

Alternatively, we *could* write a test that runs conditionally on the platform.